cranelift/filetests/filetests/isa/s390x/jumptable.clif

```diff
@@ -29,34 +29,33 @@ block5(v5: i32):
 ; VCode:
 ; block0:
 ;   llgfr %r3, %r2
+;   sllg %r4, %r3, 2
 ;   clgfi %r3, 3
-;   jghe label4
-;   sllg %r3, %r3, 2
```

Comment:
The generated code is slightly worse now as the shifting is done before the branch.  This probably doesn't have a very significant impact on the pipeline, but it does increase register pressure.  One alternative would be to compare the shifted value instead (assuming there are no overflow considerations).

But I guess this isn't serious enough that it should prevent merging this.

Reply:
Hmm, that's true -- I did the code motion without looking at the instruction semantics in detail to optimize. We can optimize in a followup if needed I suppose.